--- libiconv-1.14.old/tests/check-subst	2011-08-08 01:48:03.000000000 +0800
+++ libiconv-1.14/tests/check-subst	2012-03-09 20:15:20.143718800 +0800
@@ -20,31 +20,6 @@
 EOF
 cmp tmp-out tmp-ok
 
-# Test of --byte-subst with a non-ASCII substitution.
-
-if test "`(locale charmap) 2>/dev/null`" = UTF-8; then
-  cat > tmp-in <<\EOF
-Böse Bübchen
-EOF
-  $iconv $options_utf8 -f ASCII -t UTF-8 2>/dev/null < tmp-in > tmp-out
-  cat > tmp-ok <<\EOF
-B«0xc3»«0xb6»se B«0xc3»«0xbc»bchen
-EOF
-  cmp tmp-out tmp-ok
-fi
-
-if test "`(locale charmap) 2>/dev/null`" = UTF-8; then
-  cat > tmp-in <<\EOF
-Böse Bübchen
-EOF
-  $iconv $options_utf8 -f ASCII -t ISO-8859-1 2>/dev/null < tmp-in > tmp-out
-  $iconv -f ISO-8859-1 -t UTF-8 < tmp-out > tmp-out2
-  cat > tmp-ok <<\EOF
-B«0xc3»«0xb6»se B«0xc3»«0xbc»bchen
-EOF
-  cmp tmp-out2 tmp-ok
-fi
-
 # Test of --byte-subst with a very long substitution.
 
 cat > tmp-in <<\EOF
@@ -78,20 +53,6 @@
 EOF
 cmp tmp-out tmp-ok
 
-# Test of --unicode-subst with a non-ASCII substitution.
-
-if test "`(locale charmap) 2>/dev/null`" = UTF-8; then
-  cat > tmp-in <<\EOF
-Russian (Русский)
-EOF
-  $iconv $options_utf8 -f UTF-8 -t ISO-8859-1 2>/dev/null < tmp-in > tmp-out
-  $iconv -f ISO-8859-1 -t UTF-8 < tmp-out > tmp-out2
-  cat > tmp-ok <<\EOF
-Russian («U+0420»«U+0443»«U+0441»«U+0441»«U+043A»«U+0438»«U+0439»)
-EOF
-  cmp tmp-out2 tmp-ok
-fi
-
 # Test of --unicode-subst with a very long substitution.
 
 cat > tmp-in <<\EOF
@@ -115,31 +76,5 @@
   cmp tmp-out tmp-ok
 fi
 
-# Test of --widechar-subst:
-# wcrtomb() doesn't exist on FreeBSD 4.0 and is broken on MacOS X 10.3.
-# So far this has been tested only on a glibc system with !__STDC_ISO_10646__.
-
-if false && test "`(locale charmap) 2>/dev/null`" = UTF-8; then
-  cat > tmp-in <<\EOF
-Russian (Русский)
-EOF
-  $iconv -f char -t wchar_t < tmp-in > tmp-inw
-  LC_ALL=C                $iconv $options_ascii -f wchar_t -t ASCII < tmp-inw > tmp-out1
-  LC_ALL=de_DE.ISO-8859-1 $iconv $options_ascii -f wchar_t -t ASCII < tmp-inw > tmp-out2
-  cat > tmp-ok <<\EOF
-Russian (<00000420><00000443><00000441><00000441><0000043a><00000438><00000439>)
-EOF
-  cmp tmp-out1 tmp-ok
-  cmp tmp-out2 tmp-ok
-  if test "`(LC_ALL=de_DE.ISO-8859-1 locale charmap) 2>/dev/null`" = ISO-8859-1; then
-    options_latin1=`echo " $options_utf8" | $iconv -f UTF-8 -t ISO-8859-1`
-    LC_ALL=de_DE.ISO-8859-1 $iconv $options_latin1 -f wchar_t -t UTF-8 < tmp-inw > tmp-out1
-    cat > tmp-ok <<\EOF
-Russian («00000420»«00000443»«00000441»«00000441»«0000043a»«00000438»«00000439»)
-EOF
-    cmp tmp-out1 tmp-ok
-  fi
-fi
-
 rm -f tmp-in* tmp-out* tmp-ok
 exit 0
