--- libxml2-2.7.8/configure.in	2010-11-05 01:01:19.000000000 +0800
+++ libxml2-2.7.8-new/configure.in	2012-02-06 19:06:18.639227900 +0800
@@ -1436,6 +1436,7 @@
  WIN32_EXTRA_LIBADD="-lws2_32"
  WIN32_EXTRA_LDFLAGS="-no-undefined"
  AC_DEFINE([_WINSOCKAPI_],1,[Using the Win32 Socket implementation])
+ AC_DEFINE([HAVE_SNPRINTF],1,[Using _snprintf])
  AC_DEFINE([snprintf],[_snprintf],[Win32 Std C name mangling work-around])
  AC_DEFINE([vsnprintf],[_vsnprintf],[Win32 Std C name mangling work-around])
  if test "${PYTHON}" != ""
